### ISSUE TITLE
The Error "Something unexpected happened. Please try again or email help@uw.edu describing your problem." occurs when using Summary/More/Back/More

### ISSUE THREAD

**Issue by goulter** [UWIT-IAM/uw-husky-directory#165]
This may very well be a pre-existing bug as it doesn't always occur.
When performing a search (in this case a name search, but that shouldn't matter), select "Summary" for "Kind of Listing".
Click search
Click More
Click the browser's "back" button
Click More
Error occurs (see title of this issue).

This also can happen when
Click search
Click More
Click search
Click More
Error occurs (see title of this issue).

**Comment by jdiverp** [comment]
@goulter : please update this ticket with some additional information or remove the triage label. This is still an open issue and is easily reproduced.

**Comment by goulter** [comment]
Yes, its still open and easily repro'd - I know. I guess I don't understand why I should take off the triage label. Please clarify.

**Comment by jdiverp** [comment]
We had planned to just use the Triage label to indicate that more information needed to be added to the ticket before someone could start working on the issue and that information should be added close to the time the issue is created. Tagging @jprosser in case he has additional insights on using "Triage".

**Comment by goulter** [comment]
Oh, I didn't know that. All I knew/know is triage meant it is an open issue. All I need is to time to investigate this, as it is outside the trivial realm, at least for me.

**Comment by jprosser** [comment]
Generally we shouldn't have those Triage labels hanging around as that label indicates the issue either isn't sitting in the correct location yet, or is ambiguous in some way such that action can't be taken to resolve.

**Comment by soufianerafik** [comment]
@goulter , @jprosser , @jdiverp : 

I've investigated this issue and found the root cause - we have a double encoding problem with person hrefs when using browser back navigation.

Current behavior:
1. Initial search: href gets base64 encoded correctly
2. User views person details
3. User clicks back
4. New search: the already-encoded href gets encoded again, causing the URL parsing error

The issue occurs in the Person model's [b64_encode_href](https://github.com/UWIT-IAM/uw-husky-directory/blob/ef5ea2b24db9a5282384dc1445ad180df052af47/husky_directory/models/search.py#L222) validator which is unconditionally encoding all hrefs without checking if they're already encoded.

I have a fix ready for this issue. I'll submit a PR soon.
